src/lib/pages/backoffice/Actions/CheckIn/ItemsSearch/state/actions.js

```diff
@@ -0,0 +1,174 @@
+export const CHECKIN_IS_LOADING = 'itemCheckin/IS_LOADING';
+export const CHECKIN_SUCCESS = 'itemCheckin/SUCCESS';
+export const CHECKIN_HAS_ERROR = 'itemCheckin/HAS_ERROR';
+export const CLEAR = 'itemCheckin/CLEAR';
+export const CLEAR_SEARCH = 'itemCheckin/CLEAR_SEARCH';
+
+export const checkin = barcode => {
+  return async dispatch => {
+    dispatch({
+      type: IS_LOADING,
+    });
+
+    try {
+      const response = await itemApi.list(
+        itemApi
+          .query()
+          .withBarcode(barcode)
+          .qs()
+      );
+
+      dispatch(fetchItemLoans(response.data.hits[0].metadata.pid));
+
+      dispatch({
+        type: SUCCESS,
+        payload: response.data,
+      });
+    } catch (error) {
+      if (!error.response) {
+        dispatch(
+          addNotification('Error', `Barcode ${barcode} doesn't exist.`, 'error')
+        );
+      }
+      dispatch({
+        type: HAS_ERROR,
+        payload: error,
+      });
+      dispatch(sendErrorNotification(error));
+    }
+  };
+};
+
+export const updateQueryString = queryString => {
+  return dispatch => {
+    dispatch({
+      type: QUERY_STRING_UPDATE,
+      queryString: queryString,
+    });
+  };
+};
+
+export const fetchItemLoans = itemPid => {
+  return async dispatch => {
+    dispatch({
+      type: LOAN_IS_LOADING,
+    });
+
+    try {
+      let response = await loanApi.list(
+        loanApi
+          .query()
+          .withItemPid(itemPid)
+          .withState('ITEM_ON_LOAN')
+          .qs()
+      );
+
+      if (response.data.hits.length > 1) {
+        response.data.hits.forEach(item => {
+          item.metadata.start_date = toShortDate(item.metadata.start_date);
+          item.metadata.end_date = toShortDate(item.metadata.end_date);
+        });
+
+        dispatch({
+          type: MULTIPLE_LOAN_RESULTS,
+          payload: response.data,
+        });
+      } else {
+        const metadata = response.data.hits[0].metadata;
+        const checkinUrl = response.data.hits[0].availableActions.checkin;
+
+        dispatch(
+          checkInLoan(
+            checkinUrl,
+            metadata.document_pid,
+            metadata.patron_pid,
+            metadata.item_pid
+          )
+        );
+        dispatch({
+          type: LOAN_SUCCESS,
+          payload: response.data,
+        });
+      }
+    } catch (error) {
+      if (!error.response) {
+        dispatch(
+          addNotification('Error', `No loan for item ${itemPid}.`, 'error')
+        );
+      }
+      dispatch({
+        type: LOAN_HAS_ERROR,
+        payload: error,
+      });
+      dispatch(sendErrorNotification('Error'));
+    }
+  };
+};
+
+export const checkInLoan = (checkinUrl, documentPid, patronPid, itemPid) => {
+  return async dispatch => {
+    dispatch({
+      type: CHECKIN_IS_LOADING,
+    });
+    try {
+      let response = await documentApi.get(documentPid);
+      const pendingLoans = response.data.metadata.circulation.pending_loans;
```

Comment:
In my mind, first I would perform the action of `check-in`, so:
```
await loanApi.doAction(checkinUrl, documentPid, patronPid, {
        itemPid: itemPid,
      });
```
I would probably then have to wait for ES to index the changes:
```
await delay();
```

Then, if all previous actions success, I would query loans (pseudo-code):
```
hasLoanRequests = await loan query -> withState(PENDING) -> documentPid(documentPid) -> .total > 0
dispatch({
        type: CHECKIN_SUCCESS,
        payload: hasLoanRequests,
      });
```
If there was an error, I would even do this extra query.

But I might miss something....

Reply:
I am first getting the pending loans and then checking in the loan so that the circulation in the returned items will have the name of the previous person who loaned the document and I can display it in the list of the checked in loans.

Reply:
I think it would be great if we discuss this tomorrow, just to align on the solution with everyone!

Reply:
```
const item = await itemApi.get(itemPid.value);

await loanApi.doAction(checkinUrl, documentPid, patronPid, {
        itemPid: itemPid,
      });
await delay();

const hasLoanRequests = await loan query -> withState(PENDING) -> documentPid(documentPid) -> .total > 0
```